tests/lib/linter/linter.js

```diff
@@ -10221,507 +9939,6 @@ describe("Linter with FlatConfigArray", () => {
-                                            spy = sinon.spy(node => {
-                                                assert.strictEqual(context.getSource(node, 2), "= 6 * 7");
-                                            });
-                                            return { BinaryExpression: spy };
-                                        }
-                                    }
-                                }
-                            }
-                        },
-                        rules: { "test/checker": "error" }
-                    };
-
-                    linter.verify(code, config);
-                    assert(spy && spy.calledOnce);
-                });
-
-                it("should retrieve all text plus one character after for binary expression", () => {
-                    let spy;
-
-                    const config = {
-                        plugins: {
-                            test: {
-                                rules: {
-                                    checker: {
-                                        create(context) {
-                                            spy = sinon.spy(node => {
-                                                assert.strictEqual(context.getSource(node, 0, 1), "6 * 7;");
-                                            });
-                                            return { BinaryExpression: spy };
-                                        }
-                                    }
-                                }
-                            }
-                        },
-                        rules: { "test/checker": "error" }
-                    };
-
-                    linter.verify(code, config);
-                    assert(spy && spy.calledOnce);
-                });
-
-                it("should retrieve all text plus two characters before and one character after for binary expression", () => {
-                    let spy;
-
-                    const config = {
-                        plugins: {
-                            test: {
-                                rules: {
-                                    checker: {
-                                        create(context) {
-                                            spy = sinon.spy(node => {
-                                                assert.strictEqual(context.getSource(node, 2, 1), "= 6 * 7;");
-                                            });
-                                            return { BinaryExpression: spy };
-                                        }
-                                    }
-                                }
-                            }
-                        },
-                        rules: { "test/checker": "error" }
-                    };
-
-                    linter.verify(code, config);
-                    assert(spy && spy.calledOnce);
-                });
-
-            });
-
-            describe("context.getAncestors()", () => {
```

Comment:
Tests for `context.getAncestors()` shouldn't be removed in this PR?

Reply:
It looks like `getAncestors()`, `getScope()`, `getDeclaredVariables()`, and `markVariableAsUsed()` don't actually have a separate issue tracking their removal, despite us [announcing it](https://eslint.org/blog/2023/09/preparing-custom-rules-eslint-v9/).

So maybe I should remove those in this PR too?

Reply:
It's a subtask of https://github.com/eslint/eslint/issues/16999 (Phase 8: Removal of old APIs; just it seems that `markVariableAsUsed()` was accidentally omitted from that list, and note that the list includes removing `context.parserServices` as well) which is tracked in the v9 project.

Regardless, seems best and easiest for maintaining the alpha release to remove them as well in this PR 👍

Reply:
Ah good call, those subtasks are tricky to remember. I've updated the list to reflect reality and will add the rest into this PR.